src/osd/ReplicatedPG.cc

```diff
@@ -1627,16 +1627,29 @@ void ReplicatedPG::do_op(OpRequestRef& op)
     return;
   }
 
+  hobject_t head(m->get_oid(), m->get_object_locator().key,
+		 CEPH_NOSNAP, m->get_pg().ps(),
+		 info.pgid.pool(), m->get_object_locator().nspace);
+
   // object name too long?
-  unsigned max_name_len = MIN(g_conf->osd_max_object_name_len,
-                              osd->osd->store->get_max_object_name_length());
-  if (m->get_oid().name.size() > max_name_len) {
-    dout(4) << "do_op '" << m->get_oid().name << "' is longer than "
-            << max_name_len << " bytes" << dendl;
+  uint64_t name_length = m->get_oid().name.size() +
+    m->get_object_locator().key.size() +
+    m->get_object_locator().nspace.size();
+  if (name_length > g_conf->osd_max_object_name_len) {
+    dout(4) << "do_op name+namespace+locator is longer than "
+            << g_conf->osd_max_object_name_len
+	    << " bytes" << dendl;
```

Comment:
I don't think this hunk is necessary if the ObjectStore has its own check.  Also, I don't think it makes sense.  If the max length is 16, then "0123456789abcdef" is a valid name.  I should be able to specify an equally long key so that I can make "foo" stored next to it...